gcloud-java-contrib/gcloud-java-nio/src/main/java/com/google/gcloud/storage/contrib/nio/CloudStorageFileSystemProvider.java

```diff
@@ -195,24 +219,30 @@ private SeekableByteChannel newWriteChannel(
             throw new UnsupportedOperationException(option.toString());
         }
       } else if (option instanceof CloudStorageOption) {
-        // These will be interpreted later.
+        // XXX: We need to interpret these later
       } else {
         throw new UnsupportedOperationException(option.toString());
       }
     }
-    CloudStoragePath cloudPath = checkPath(path);
-    if (cloudPath.seemsLikeADirectoryAndUsePseudoDirectories()) {
-      throw new CloudStoragePseudoDirectoryException(cloudPath);
+
+    if (!metas.isEmpty()) {
+      infoBuilder.metadata(metas);
+    }
+    if (!acls.isEmpty()) {
+      infoBuilder.acl(acls);
     }
-    if (wantCreateNew) {
-      // XXX: Java's documentation says this should be atomic.
-      if (gcsService.getMetadata(cloudPath.getGcsFilename()) != null) {
-        throw new FileAlreadyExistsException(cloudPath.toString());
+
+    try {
+      return new CloudStorageWriteChannel(gcsStorage.writer(infoBuilder.build(),
+          writeOptions.toArray(new Storage.BlobWriteOption[0])));
+    } catch (StorageException oops) {
+      // RPC API can only throw StorageException, but CloudStorageFileSystemProvider
+      // can only throw IOException. So we tunnel the latter inside the former.
+      if (oops.getCause() instanceof IOException) {
```

Comment:
I'm not sure if this is the behavior we want. We lose information when we peel out the cause. Like a stack trace to whatever part of gcloud-java-storage threw this. We're probably better off throwing a new `IOException` that wraps `oops`. That way people can check the cause of our exception object, see if it's a `BaseServiceException` and if so, check its `code()` and `reason()` and stuff.